src/video/encoder.hpp

```diff
@@ -0,0 +1,36 @@
+#ifndef ENCODER_HPP_
+#define ENCODER_HPP_
```

Comment:
Also, since we're making this a file in our own repo, let's get rid of the `ifndef` guard and use `#pragma once` instead, since that's consistent with what we do everywhere else.

So to clarify, remove the first two lines (the `ifndef` and `define`) and also the `endif` on the last line, and just put `#pragma once` at the top of the file.